hudi-client/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -145,7 +145,8 @@ public Boolean shouldAssumeDatePartitioning() {
   }
 
   public Integer getTimelineLayoutVersion() {
-    return Integer.parseInt(props.getProperty(TIMELINE_LAYOUT_VERSION));
+    return props.containsKey(TIMELINE_LAYOUT_VERSION)
+        ? Integer.parseInt(props.getProperty(TIMELINE_LAYOUT_VERSION)) : null;
```

Comment:
should we just use a VERSION_0 instead of `null`?

Reply:
and can't we do this using the regular defaults way? why the special handling for containsKey etc? `props.getProperty(k, default)`?

Reply:
@vinothchandar : Unlike other configs, this is a config defined at 2 levels - hoodie.properties and in HoodieWriteConfig. If we actually keep a non-null value in config, the semantics is that it would override the one in hoodie.properties. If we keep version_0 as default, it would override the version even for new tables which has Version_1 in hoodie.properties ?   Let me know if I am missing something.

Reply:
Main issue here seems to be that, we want this config to be respected at a write level, even when the value is not in `hoodie.properties`, for e.g support timeline version =1 even for older tables. `hoodie.properties` is involved here, purely because the query (any timeline reader) needs to know how to read the timeline correct? 

code style aside, I feel this is still not very clear to me.. :(